pkg/utils/nodeutils/predicates.go

```diff
@@ -24,3 +29,15 @@ func IsReadyAndSchedulable(node v1.Node) bool {
 	}
 	return false
 }
+
+func IsPastTTL(node *v1.Node) (bool, error) {
+	ttl, ok := node.Annotations[v1alpha1.ProvisionerTTLKey]
+	if !ok {
+		return false, nil
+	}
+	ttlTime, err := time.Parse(time.RFC3339, ttl)
+	if err != nil {
```

Comment:
Can we remove the error case entirely, and just log and return false. I think callers would have cleaner code if this weren't errorable, and instead were just a node predicate `(node) => bool`

Reply:
I agree. That makes sense.